common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -37,49 +44,56 @@ def initialize(github_redirection_service:)
         end
 
         def sanitize_links_and_mentions(text:)
-          # We don't want to sanitize any links or mentions that are contained
-          # within code blocks, so we split the text on "```" or "~~~"
-          lines = []
-          scan = StringScanner.new(text)
-          until scan.eos?
-            line = scan.scan_until(CODEBLOCK_REGEX) ||
-                   scan.scan_until(EOS_REGEX)
-            delimiter = line.match(CODEBLOCK_REGEX)&.to_s
-            unless delimiter && lines.count { |l| l.include?(delimiter) }.odd?
-              line = sanitize_mentions(line)
-            end
-            lines << line
-          end
+          doc = CommonMarker.render_doc(
+            text, :LIBERAL_HTML_TAG, COMMONMARKER_EXTENSIONS
+          )
 
-          sanitize_links(lines.join)
+          sanitize_mentions(doc)
+          sanitize_links(doc)
+          doc.to_html(COMMONMARKER_OPTIONS, COMMONMARKER_EXTENSIONS)
         end
 
         private
 
-        def sanitize_mentions(text)
-          text.gsub(%r{(?<![A-Za-z0-9`~])@#{GITHUB_USERNAME}/?}) do |mention|
-            next mention if mention.end_with?("/")
+        def sanitize_mentions(doc)
+          mention_regex = %r{(?<![A-Za-z0-9`~])@#{GITHUB_USERNAME}/?}
 
-            last_match = Regexp.last_match
-            sanitized_mention = mention.gsub("@", "@&#8203;")
+          doc.walk do |node|
+            if node.type == :text && node.string_content.match?(mention_regex)
+              nodes = []
+              scan = StringScanner.new(node.string_content)
+
+              until scan.eos?
+                line = scan.scan_until(mention_regex) ||
+                       scan.scan_until(EOS_REGEX)
+                mention = line.match(mention_regex)&.to_s
```

Comment:
Looks like there was a method on scan to get this
```suggestion
                mention = scan.matched
```